src/MqttMailingService.cpp

```diff
@@ -56,6 +56,14 @@ MqttMailingService::startWithDelegatedWiFi(const char* ssid, const char* pass) {
     WiFi.begin(ssid, pass);
     start();
 
+    // Wait for Wi-Fi connection, since MQTT mailing service
+    // can't establish a connection without Wi-Fi.
+    while (!WiFi.isConnected()) {
+        ESP_LOGD(MqttMailingService::TAG, "Waiting for Wi-Fi connection...");
+        sleep(1);
+    }
```

Comment:
As discussed I prefer to have an option for the user to start the mailing service without a blocking call waiting for the WiFi.
Furthermore we if we wait for a WiFi connection we should do so before calling start() for the mqtt initialization to avoid the error messages due to missing WiFi config

Reply:
I now added an optional parameter (defaults to non-blocking), tell me what you think.